### PR TITLE
OCM-9596 | test: fix ids:72162,64620 fix failures on PROW CI testing

### PR DESCRIPTION
**Pull request by yuwang-RH**
[OCM-9596](https://issues.redhat.com//browse/OCM-9596) | test: fix ids:72162,64620 fix failures on PROW CI testing

72162: use '-d' as the original '-D' only supports on MacOS
64620: Add function and to run command in manual mode in tmp dir to avoid the failure of missing permission

Logs are here,https://privatebin.corp.redhat.com/?6de329c37662c528#DLdY23pjwPy7gV13Lh8tP2B5KDgfaTkbHG4rPgbXmZHr

### CODE REVIEW

File: tests/e2e/test_rosacli_oidc_config.go [hunk 1]
```diff
@@ -249,9 +266,17 @@ var _ = Describe("Register ummanaged oidc config testing",
 			Expect(installerRole).ToNot(BeNil())
 			roleArn := installerRole.RoleArn
 
+			By("Create a temp dir to execute the create commands")
+			output, err := rosaClient.Runner.RunCMD([]string{"mktemp", "-d"})
+			Expect(err).To(BeNil())
+			dirToClean = strings.TrimSpace(output.String())
```
Comment (yasun1) [inline tests/e2e/test_rosacli_oidc_config.go:272]: A common function to create the temporary file could be appreciated.

File: tests/e2e/test_rosacli_oidc_config.go [hunk 2]
```diff
@@ -202,16 +202,26 @@ var _ = Describe("Register ummanaged oidc config testing",
...
 		BeforeEach(func() {
 			By("Init the client")
 			rosaClient = rosacli.NewClient()
 			ocmResourceService = rosaClient.OCMResource
-			rosaClient = rosacli.NewClient()
+
+			By("Get the default dir")
+			output, err := rosaClient.Runner.RunCMD([]string{"pwd"})
+			Expect(err).To(BeNil())
+			defaultDir = strings.TrimSpace(output.String())
```
Comment (yasun1) [inline tests/e2e/test_rosacli_oidc_config.go:217]: Why not use `err := os.Getwd()`?

File: tests/e2e/test_rosacli_oidc_config.go [hunk 3]
```diff
@@ -228,6 +238,13 @@ var _ = Describe("Register ummanaged oidc config testing",
 				"--prefix", accountRolePrefix,
 				"-y")
 			Expect(err).To(BeNil())
+
+			By("Delete temp dir")
+			if dirToClean != "" {
+				_, err := rosaClient.Runner.RunCMD([]string{"rm", "-rf", dirToClean})
+				Expect(err).To(BeNil())
+			}
```
Comment (yasun1) [inline tests/e2e/test_rosacli_oidc_config.go:246]: Why not use `err := os.RemoveAll(tempfile)`?

File: tests/e2e/test_rosacli_oidc_config.go (hunk 1 above)
Comment (yuwang-RH) [reply tests/e2e/test_rosacli_oidc_config.go:263]: I will add the common function if other place needs that in future

File: tests/e2e/test_rosacli_oidc_config.go (hunk 2 above)
Comment (yuwang-RH) [reply tests/e2e/test_rosacli_oidc_config.go]: Updated. Thanks

File: tests/e2e/test_rosacli_oidc_config.go (hunk 3 above)
Comment (yuwang-RH) [reply tests/e2e/test_rosacli_oidc_config.go]: After discussion, remove this step to leave the temp dir cleanup by job itself

**Comment by yuwang-RH** [comment]
@yasun1 Updated in the latest commit also include littile change to skip the error check for Getwd as we can trust that basic function and avoid probable permission issue for the default dir. Could you help take a look the latest commit?Thanks

**Pull request merged**